src/index/Index.h

```diff
@@ -3,60 +3,12 @@
 // Author: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
 #pragma once
 
-#include <array>
-#include <fstream>
-#include <google/sparse_hash_set>
-#include <memory>
-#include <nlohmann/json.hpp>
-#include <optional>
-#include <string>
-#include <stxxl/vector>
-#include <vector>
-#include "../engine/ResultTable.h"
+#include "../engine/IdTable.h"
```

Comment:
In this file,
The signatures of all public methods of the original index class should be unchanged,
but only the declarations should be present here (The Pimpl-Idiom).